x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -224,10 +223,14 @@ export const transformOrImportError = (
   }
 };
 
-export const getDuplicates = (lodashDict: Dictionary<number>): string[] => {
-  const hasDuplicates = Object.values(lodashDict).some(i => i > 1);
+export const getDuplicates = (ruleDefinitions: RuleAlertParamsRest[], by: 'rule_id'): string[] => {
+  const mappedDuplicates = countBy(
+    by,
+    ruleDefinitions.filter(r => r[by] != null)
+  );
+  const hasDuplicates = Object.values(mappedDuplicates).some(i => i > 1);
   if (hasDuplicates) {
-    return Object.keys(lodashDict).filter(key => lodashDict[key] > 1);
+    return Object.keys(mappedDuplicates).filter(key => mappedDuplicates[key] > 1);
```

Comment:
cc @dhurley14 -- @XavierM and I found a bug here where duplicating rules would fail as we're using the `bulk_create` API and it would mark all the rules as dupes since they all have `rule_id: undefined` and then wouldn't create them. I've created https://github.com/elastic/kibana/issues/58287 for providing an explicit API for duplicating rules.

Reply:
I talked to @dhurley14 about it and show the code and he was ok with the change

Reply:
Yes thanks for this! We could try and add an e2e test for this case in the future. Thanks @XavierM !